plasma-mode.md

```diff
@@ -0,0 +1,203 @@
+The DA server returns a commiment to the data to the batcher & the batcher posts that commitment to L1.
+The op-node performs derivation with the modification that when it finds a commiment on L1, it asks its DA server to get the data associated with the commitment.
+Then the rest of the derivation pipeline functions as normal.
+Finalization is modified because a L2 block can only be finalized when the commitment that was used to derive it can no longer be challenged.
+Commitment finalization takes significantly longer in Plasma Mode because of the challenge period.
+Data from commitments is also used prior to knowing if the commitment is valid or not.
+
+# Proposed Solution
+
+We propose a generic commitment type which can be interpreted by a DA server to fetch & validate the data. Challenge commitments is done in per DA layer smart contract where the smart contract emits events about the status of the challenge. The op-node then interprets those events to handle commitment validity and finality.
+
+## DA Server Spec
+
+The DA Server is [specified here](https://github.com/ethereum-optimism/specs/blob/main/specs/experimental/plasma.md#da-server).
+At a high level, the DA server implements a HTTP `get` that the op-node uses to fetch data and a HTTP `put` that the batcher uses to post data.
+The DA Server is responsible for verifying that the data returned from the DA layer matches the commitment.
+
+
+### Alternatives
+
+We considered moving the Data Fetching logic into the op-node. It would be possible to place it the fetcher component behind a Go interface; however we rejected this solution in favor of putting the fetching logic over a network boundary.
+Moving the logic to a dedicated HTTP server means that the same op-node binary can work with multiple layers.
+The other option to avoid compiling in the DA Layer specific logic is to use [go plugins](https://pkg.go.dev/plugin). We reject this solution because it requires that the DA fetching logic be implementing in Go and it add places restrictions on how the resulting binary may be used.
+
+## Generic Commitment
+
+The generic commtiment is [specified here](https://github.com/ethereum-optimism/specs/blob/main/specs/experimental/plasma.md#input-commitment-submission).
+The relevant part is copied to this document:
+
+The da-service commitment is as follows: `da_layer_byte ++ payload`. The DA layer byte must be initially restricted to the range `[0, 127)`. This specification will not apportion DA layer bytes, but different DA layers should coordinate to ensure that the DA layer bytes do not conflict. DA Layers can do so in [this discussion](https://github.com/ethereum-optimism/specs/discussions/135). The payload is a bytestring which is up to the DA layer to specify. The DA server should be able to parse the payload, find the data on the DA layer, and verify that the data returned from the DA layer matches what was committed to in the payload.
```

Comment:
DA Layers often want more than 32 bytes. Usually they put in block height / hash + a cryptographic commitment to the data. Celestia is 8 bytes for the block number + 32 bytes for crypto commitment.

Reply:
@protolambda what is the benefit of a fixed-size commitment in this case? What does it buy us?

@trianglesphere if we were to go with a fixed-size commitment, while 32 bytes is too small for current use cases, couldn't we set a reasonable upper-bound that would be (e.g., 64 bytes)?

Reply:
A 32-byte identifier would be easier to work with in the EVM, and bound what the op-node has to deal with. The pre-image of it can still be arbitrarily large if we want, and that's what is posted to the L1 inbox I imagine.